core/stats.go

```diff
@@ -0,0 +1,101 @@
+package core
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/evcc-io/evcc/server/db"
+	"github.com/evcc-io/evcc/util"
+)
+
+// publisher gives access to the site's publish function
+type publisher interface {
+	publish(key string, val interface{})
+}
+
+// Publishes long term charging statistics
+type Stats struct {
+	updated time.Time // Time of last charged value update
+	log     *util.Logger
+}
+
+func NewStats() *Stats {
+	return &Stats{
+		log: util.NewLogger("stats"),
+	}
+}
+
+// Update publishes stats for the last 30 and 365 days
+func (s *Stats) Update(p publisher) {
+	if time.Since(s.updated) < time.Hour {
+		return
+	}
+
+	s.publish(30, p)
+	s.publish(365, p)
```

Comment:
add today/ current week/ month/ year?

Reply:
I've added "total".

For this view I'd prefer to stick with fixed timeframes and not go to calender day/week/month. I'd see those in the charging session overview in addition to the tabular data.

Another more technical reason: These stats are based on finished charging sessions. So a today view (or first day of month) would show no data if the first charging session has not finished yet. For broader time frames this is not an issue.